app/models/manageiq/providers/redfish/inventory/collector/physical_infra_manager.rb

```diff
@@ -3,5 +3,23 @@ class Inventory::Collector::PhysicalInfraManager < Inventory::Collector
     def physical_servers
       rf_client.Systems.Members.collect(&:raw)
     end
+
+    def physical_server_details
+      rf_client.Systems.Members.collect { |s| get_server_location(s) }
+    end
+
+    private
+
+    def get_server_location(server)
+      chassis = [server.Links.Chassis[0]]
```

Comment:
Is it possible for `server.Links.Chassis` to be nil or an empty array?  If so you might want something like `chassis = Array(server.Links.Chassis&.first)`

Reply:
`server.Links.Chassis` cannot be `nil` and it should have only one element in properly configured services. That being said, assuming properly configured services might be a bit naive.

I will rework this part of the code to take into account those edge cases.